hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/discovery/DiscoveryAddressProvider.java

```diff
@@ -41,7 +43,8 @@ public DiscoveryAddressProvider(DiscoveryService discoveryService, LoggingServic
 
     @Override
     public Collection<InetSocketAddress> loadAddresses() {
-        Iterable<DiscoveryNode> discoveredNodes = discoveryService.discoverNodes();
+        Iterable<DiscoveryNode> discoveredNodes = checkNotNull(discoveryService.discoverNodes(),
```

Comment:
after this pr, we may return empty list rather than `null` https://github.com/hazelcast/hazelcast-azure/blob/master/src/main/java/com/hazelcast/azure/AzureDiscoveryStrategy.java#L144
// cc @mesutcelik @emre-aydin